packages/smart-contracts/src/contracts/ERC20EscrowToPayV1.sol

```diff
@@ -0,0 +1,385 @@
+        require(invoiceMapping[_paymentRef].paymentToken.transferFrom(
+            invoiceMapping[_paymentRef].payer,
+            address(this), 
+            (invoiceMapping[_paymentRef].amount + invoiceMapping[_paymentRef].feeAmount)), 
+            "_deposit: Can't _deposit tokens to Escrow, did you approve ERC20 token first?"
+        );
+        return true; 
+    }
+
+    /**
+     * @notice Withdraw the funds from the escrow.  
+     * @param _paymentRef Reference of the related Invoice.
+     * @dev Internal function to pay fees and withdraw funds to a given reciever.
+     */
+    function _withdraw(bytes memory _paymentRef, address _receiver) internal returns (bool result) {
+        /// Normal flow, funds are transfered to payee.
+        if (invoiceMapping[_paymentRef].amount != 0 ) {
+            uint256 _amount = invoiceMapping[_paymentRef].amount;
+            invoiceMapping[_paymentRef].amount = 0;
+            /// Give approval to transfer from ERC20EscrowToPayV1 => ERC20FeeProxy contract.
+            invoiceMapping[_paymentRef].paymentToken.approve(address(paymentProxy), _amount + invoiceMapping[_paymentRef].feeAmount);
+    
+            // Pay the invoice request and fees
+            paymentProxy.transferFromWithReferenceAndFee(
+                address(invoiceMapping[_paymentRef].paymentToken),
+                _receiver,
+                _amount, 
+                _paymentRef, 
+                invoiceMapping[_paymentRef].feeAmount, 
+                payable(invoiceMapping[_paymentRef].feeAddress)
+            );  
+            return true;
+        }
+        /// Resolve dispute flow, Escrowed funds are transfered to payee.
+        if (disputeMapping[_paymentRef].amount != 0) {
+            uint256 _amount = disputeMapping[_paymentRef].amount;
+            /// reset the amount to prevent reentrency attack.
+            disputeMapping[_paymentRef].amount = 0;
+            
+            disputeMapping[_paymentRef].paymentToken.approve(address(paymentProxy), _amount + disputeMapping[_paymentRef].feeAmount);
+
+            // Pay the request and fees
+            paymentProxy.transferFromWithReferenceAndFee(
+                address(disputeMapping[_paymentRef].paymentToken),
+                _receiver, 
+                _amount, 
+                _paymentRef, 
+                disputeMapping[_paymentRef].feeAmount,
+                payable(disputeMapping[_paymentRef].feeAddress)
+            );
+            return true;
+        }
+
+        revert WithdrawFailed("Withdraw FAILED");  
+    } 
+
+    /**
+     * @notice Used to change the feeAmount and feeAddress of any escrow.
+     * @param _paymentRef Reference of the related Invoice.
+     * @param _feeAmount The amount of fee to pay.
+     * @param _feeAddress The fee recipient.
+     * @dev Housekeeping.
+     */
+    function changeFeeAndAddress(bytes memory _paymentRef, uint _feeAmount, address _feeAddress) 
```

Comment:
I don't understand this ? I think the most important is to let the payer change the payee address, not the fees